site/src/contexts/useAgenticChat.ts

```diff
@@ -0,0 +1,23 @@
+import { experiments } from "api/queries/experiments";
+
+import { useEmbeddedMetadata } from "hooks/useEmbeddedMetadata";
+import { useEffect, useState } from "react";
+import { useQuery } from "react-query";
+
+interface AgenticChat {
+	readonly enabled: boolean;
+}
+
+export const useAgenticChat = (): AgenticChat => {
+	const { metadata } = useEmbeddedMetadata();
+	const enabledExperimentsQuery = useQuery(experiments(metadata.experiments));
+
+	const [enabled, setEnabled] = useState<boolean>(false);
+	useEffect(() => {
+		if (enabledExperimentsQuery.data?.includes("agentic-chat")) {
+			setEnabled(true);
+		}
+	});
+
+	return { enabled };
```

Comment:
There is no need for enabled to be a state since it is value is just a computation. 

```ts
return { enabled: enabledExperimentsQuery.data?.includes("agentic-chat") }
```